pkg/hwinfo/hwinfo.go

```diff
@@ -1,16 +1,48 @@
 package hwinfo
 
 import (
-	"errors"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
 )
 
-var ErrNotPresent = errors.New("hwinfo: not present")
+type pciDeviceInfo struct {
+	Address     string `json:"address"`
+	DeviceType  string `json:"device_type,omitempty"`
+	VendorName  string `json:"vendor_name,omitempty"`
+	ProductName string `json:"product_name"`
+	Description string `json:"description,omitempty"`
+}
+
+type usbDeviceInfo struct {
+	Address     string `json:"address,omitempty"`
+	VendorName  string `json:"vendor_name,omitempty"`
+	DeviceID    string `json:"id"`
+	Description string `json:"description,omitempty"`
+}
+
+type monitorInfo struct {
+	ID          string `json:"id"`
+	Description string `json:"description,omitempty"`
+	VendorName  string `json:"vendor_name,omitempty"`
+	Size        string `json:"size,omitempty"`
+	Resolution  string `json:"resolution,omitempty"`
+}
 
 func Inventory() (map[string]interface{}, error) {
 	hw, err := fetchInventory()
 	if err != nil {
-		return nil, err
+		err = errors.Wrap(err, "hwinfo")
+		log.Error(err)
+		return hw, err
 	}
 
 	return hw, nil
 }
+
+func mergeStringMaps(mapA, mapB *map[string]interface{}) map[string]interface{} {
+	result := *mapA
```

Comment:
Not sure why this is here instead just map-by-value. It wouldn't create a copy anyway.
https://play.golang.org/p/9bFI7agWcJ3

Reply:
fixed